tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
-    """Test the light module."""
-
-    # pylint: disable=invalid-name
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-        user_light_file = self.hass.config.path(light.LIGHT_PROFILES_FILE)
-
-        if os.path.isfile(user_light_file):
-            os.remove(user_light_file)
-
-    def test_methods(self):
-        """Test if methods call the services as expected."""
-        # Test is_on
-        self.hass.states.set("light.test", STATE_ON)
-        assert light.is_on(self.hass, "light.test")
-
-        self.hass.states.set("light.test", STATE_OFF)
-        assert not light.is_on(self.hass, "light.test")
-
-        # Test turn_on
-        turn_on_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TURN_ON)
-
-        common.turn_on(
-            self.hass,
-            entity_id="entity_id_val",
-            transition="transition_val",
-            brightness="brightness_val",
-            rgb_color="rgb_color_val",
-            xy_color="xy_color_val",
-            profile="profile_val",
-            color_name="color_name_val",
-            white_value="white_val",
-        )
+from tests.common import async_mock_service
 
-        self.hass.block_till_done()
 
-        assert 1 == len(turn_on_calls)
-        call = turn_on_calls[-1]
+@pytest.fixture
+def mock_storage(hass, hass_storage):
+    """Clean up user light files at the end."""
+    yield
+    user_light_file = hass.config.path(light.LIGHT_PROFILES_FILE)
 
-        assert light.DOMAIN == call.domain
-        assert SERVICE_TURN_ON == call.service
-        assert "entity_id_val" == call.data.get(ATTR_ENTITY_ID)
-        assert "transition_val" == call.data.get(light.ATTR_TRANSITION)
-        assert "brightness_val" == call.data.get(light.ATTR_BRIGHTNESS)
-        assert "rgb_color_val" == call.data.get(light.ATTR_RGB_COLOR)
-        assert "xy_color_val" == call.data.get(light.ATTR_XY_COLOR)
-        assert "profile_val" == call.data.get(light.ATTR_PROFILE)
-        assert "color_name_val" == call.data.get(light.ATTR_COLOR_NAME)
-        assert "white_val" == call.data.get(light.ATTR_WHITE_VALUE)
+    if os.path.isfile(user_light_file):
+        os.remove(user_light_file)
 
-        # Test turn_off
-        turn_off_calls = mock_service(self.hass, light.DOMAIN, SERVICE_TURN_OFF)
 
-        common.turn_off(
-            self.hass, entity_id="entity_id_val", transition="transition_val"
-        )
+async def test_methods(hass):
+    """Test if methods call the services as expected."""
+    # Test is_on
+    hass.states.async_set("light.test", STATE_ON)
+    assert light.is_on(hass, "light.test")
 
-        self.hass.block_till_done()
+    hass.states.async_set("light.test", STATE_OFF)
+    assert not light.is_on(hass, "light.test")
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)